odoo/addons/base/views/ir_model_views.xml

```diff
@@ -304,6 +304,10 @@
                                         <field name="domain" groups="base.group_no_one"
                                             attrs="{'readonly': [('ttype','not in',['many2one','one2many','many2many'])],
                                                     'invisible': [('ttype','not in',['many2one','one2many','many2many'])]}"/>
+                                        <field name="currency_field"
+                                            attrs="{'required': [('ttype','=','monetary')],
```

Comment:
required if ttype = monetary, but some monetary does have the currency stored in currency_field. Hardcoded in Monetary class as either `currency_id` or `x_currency_id`